game-core/src/main/java/games/strategy/triplea/delegate/battle/BattleActions.java

```diff
@@ -35,4 +38,27 @@ void landParatroopers(
   void endBattle(IDelegateBridge bridge);
 
   void attackerWins(IDelegateBridge bridge);
+
+  void findTargetGroupsAndFire(
+      ReturnFire returnFire,
+      String stepName,
+      boolean defending,
+      GamePlayer firingPlayer,
+      Predicate<Unit> firingUnitPredicate,
+      Collection<Unit> firingUnits,
+      Collection<Unit> firingUnitsWaitingToDie,
+      Collection<Unit> enemyUnits,
+      Collection<Unit> enemyUnitsWaitingToDie);
+
+  void remove(
+      Collection<Unit> killedUnits,
+      IDelegateBridge bridge,
+      Territory battleSite,
+      Boolean defenderDying);
+
+  void damagedChangeInto(
+      GamePlayer player,
+      Collection<Unit> units,
```

Comment:
We're baking in an in/out parameter here and this value is modified.
What would you think if we returned instead the collection of transformed units? That would imply that the caller would be the one doing the `IDelagetBridge` work.